enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,287 @@
+ │ lsp stor-node1     │──┘     └───│ lsp stor-node2      │      
+ │ options:           │            │ options:            │    
+ │  arp_proxy:        │            │   arp_proxy:        │
+ │   0a:58:0a:f3:00:00│            │    0a:58:0a:f3:00:00│
+ │   169.254.1.1      │            │    169.254.1.1      │
+ └────────────────────┘            └─────────────────────┘
+```
+
+**VMs logical switch ports:**
+
+The logical switch port for new VMs will use ovn-k ipam to reserve an address
+and when live migrate the new lsp created on target will re-use the source 
+lsp address, since the virt-launcher pod netns address will not be set by 
+cni this is fine.
+
+Also the DHCP options will be configured to deliver the address to the VMs
+
+```text
+    ┌──────────────────────┐     ┌────────────────────┐
+    │logical switch node1  │     │logical switch node2│
+ ┌──────────────────────┐  │     │   ┌──────────────────────┐      
+ │ lsp ns-virt-launcher1│──┘     └───│ lsp ns-virt-launcher2│     
+ │ dhcpv4_options: 1234 │            │ dhcpv4_options: 1234 │
+ │ address:             │            │ address:             │    
+ │  0a:58:0a:f4:00:01   │            │  0a:58:0a:f4:00:01   │
+ │  10.244.0.8          │            │  10.244.0.8          │
+ └──────────────────────┘            └──────────────────────┘
+
+┌─────────────────────────────────┐
+│ dhcp-options 1234               │
+│   lease_time: 3500              │
+│   router: 169.254.1.1           │
+│   dns_server: [kubedns]         │
+│   server_id: 169.254.1.1        │
+│   server_mac: c0:ff:ee:00:00:01 │
+└─────────────────────────────────┘
+```
+
+**Virt-launcher pod address:**
+ The CNI will not set an address at virt-launcher pod netns, that address is
+ assigned to the VM with the DHCP options from the LSP, this allows to use 
+ kubevirt bridge binding with pod networking and still do live migration.
+
+#### IPAM
+
+The /32 routing feature allows an address to be runnig at a node different from
+the one "owning" the subnet the address is comming from. This will happend
+after VM live migration.
+
+One scenario for live migration is to shut down the node the VMs were migrated
+from, this means that the IPAM node subnet should go back to the pool but since
+the migrated VMs contains IPs from it those IPs should reserved in case the 
+subnet is assigned to a new node.
+
+On that case before assigning the subnet to the node the VMs ips need to be 
+reserved so they don't get assigned to new pods
+
+Another scenario is ovn-kubernetes pods restarting after live migration, on 
+that case ovnkube-master should discover to what ip pool the VM belongs 
+(using an annotation or searching for subnets) and reserve the address.
+
+#### Detecting hypershift VMs and live migration
+
+This enhancement only applies for hypershift kubevirt workers so the rest of 
+virtual machine should not use it, to do a new label will be added to
+virt-launcher pod so ovn-k can detect if the /32 routing need to be done.
```

Comment:
can we go ahead and give the label a name now so it's an agreed on part of the enhancement pr?

maybe something like `kubevirt.io/allow-pod-network-bridge-binding-live-migration`. that's long, but it comes in under the 63 character label key limit and is pretty descriptive to exactly what we're doing.

```
$ printf "kubevirt.io/allow-pod-network-bridge-binding-live-migration" | wc -m
59
```